packages/millicast-sdk/src/PeerConnectionStats.js

```diff
@@ -62,221 +63,62 @@ export const peerConnectionStatsEvents = {
 }
 
 export default class PeerConnectionStats extends EventEmitter {
-  constructor (peer) {
+  constructor (peer, options = { autoInitStats: true }) {
     super()
     this.peer = peer
-    this.stats = null
-    this.emitInterval = null
-    this.previousStats = null
+    this.collection = null
+    this.initialized = false
+
+    if (options.autoInitStats) {
+      this.init()
+    }
   }
 
   /**
    * Initialize the statistics monitoring of the RTCPeerConnection.
    */
   init () {
+    if (this.initialized) {
+      logger.warn('PeerConnectionStats.init() has already been called. Automatic initialization occurs when the PeerConnectionStats object is constructed.')
+      return
+    }
     logger.info('Initializing peer connection stats')
-    this.emitInterval = setInterval(async () => {
-      const stats = await this.peer.getStats()
-      this.parseStats(stats)
-      Diagnostics.addStats(this.stats)
-      /**
-       * Peer connection incoming stats.
-       *
-       * @event PeerConnection#stats
-       * @type {ConnectionStats}
-      */
-      this.emit(peerConnectionStatsEvents.stats, this.stats)
-    }, 1000)
+    const peer = this.peer
+    try {
+      this.collection = new WebRTCStats({
+        getStatsInterval: 1000,
+        getStats: () => {
+          return peer.getStats()
+        },
+        includeRawStats: true
+      })
+      this.collection.on('stats', (stats) => {
+        Diagnostics.addStats(stats)
+        this.emit(peerConnectionStatsEvents.stats, stats)
+      })
+      this.collection.start()
+    } catch (e) {
+      logger.error(e)
+    }
+    this.initialized = true
```

Comment:
should initialized be set before `this.collection.start()` ?

Reply:
Moving to try block.